kcp and extensions failing version verification

**Describe the bug**

With no kubeconfig file in `~/.kube.cofig` run:

```
kubectl kcp --help
```

The extension panics with an error:

```
$ kubectl kcp --help
panic: version string "+kcp-v0.6.0-alpha.0-129-gfedb8387047c1d" doesn't match expected regular expression: "^v(\d+\.\d+\.\d+)"

goroutine 1 [running]:
k8s.io/component-base/metrics.parseVersion({{0x35123b7, 0x0}, {0x35123b8, 0x0}, {0x35392e0, 0x27}, {0x3515c80, 0x8}, {0x3512418, 0x5}, ...})
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220721163516-e48c0ebdf7b8/metrics/version_parser.go:47 +0x274
k8s.io/component-base/metrics.newKubeRegistry({{0x35123b7, 0x0}, {0x35123b8, 0x0}, {0x35392e0, 0x27}, {0x3515c80, 0x8}, {0x3512418, 0x5}, ...})
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220721163516-e48c0ebdf7b8/metrics/registry.go:320 +0x118
k8s.io/component-base/metrics.NewKubeRegistry()
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220721163516-e48c0ebdf7b8/metrics/registry.go:335 +0x78
k8s.io/component-base/metrics/legacyregistry.init()
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220721163516-e48c0ebdf7b8/metrics/legacyregistry/registry.go:29 +0x1d
```

Same thing with `ws` and `workspaces` extensions.

A nicer error message would be better, and when just getting the help it perhaps should display the help still even when there is no kubeconfig file.

Or it could be something totally different.

So the process I have followed is to build from GitHub source code, but I have only copied select binaries from that build into a distinct container.

```
FROM fedora:36 as builder

RUN INSTALL_PKGS=" \
        go \
        make \
    " && \
    dnf install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
    dnf clean -y --enablerepo='*' all

RUN git clone https://github.com/kcp-dev/kcp.git && \
    cd kcp && \
    make install

FROM ghcr.io/xxx/xxx:latest

COPY --from=builder --chown=1001:0 /root/go/bin/kcp bin/
COPY --from=builder --chown=1001:0 /root/go/bin/kubectl-* bin/
```

The errors come from running the standalone binaries in the container. The expectation here is that none of the source code is needed and only the binaries.

Since I didn't have `kcp` running, I figured the next step was to run `kcp` itself, but that fails as well.

```
[~] $ unset KUBECONFIG
[~] $ kcp start
panic: version string "+kcp-v0.6.0-alpha.0-129-gfedb8387047c1d" doesn't match expected regular expression: "^v(\d+\.\d+\.\d+)"

goroutine 1 [running]:
k8s.io/component-base/metrics.parseVersion({{0x49eb4b0, 0x0}, {0x49eb4b1, 0x0}, {0x4a207a0, 0x27}, {0x49ef8c0, 0x8}, {0x49eb54c, 0x5}, ...})
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220721163516-e48c0ebdf7b8/metrics/version_parser.go:47 +0x274
k8s.io/component-base/metrics.newKubeRegistry({{0x49eb4b0, 0x0}, {0x49eb4b1, 0x0}, {0x4a207a0, 0x27}, {0x49ef8c0, 0x8}, {0x49eb54c, 0x5}, ...})
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220721163516-e48c0ebdf7b8/metrics/registry.go:320 +0x118
k8s.io/component-base/metrics.NewKubeRegistry()
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220721163516-e48c0ebdf7b8/metrics/registry.go:335 +0x78
k8s.io/component-base/metrics/legacyregistry.init()
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220721163516-e48c0ebdf7b8/metrics/legacyregistry/registry.go:29 +0x1d
```

Using last tagged version doesn't help either.

```
[~] $ kcp start
panic: version string "+kcp-v0.6.3" doesn't match expected regular expression: "^v(\d+\.\d+\.\d+)"

goroutine 1 [running]:
k8s.io/component-base/metrics.parseVersion({{0x49a6cda, 0x0}, {0x49a6cdb, 0x0}, {0x49ab0d8, 0xb}, {0x49ab0a8, 0x8}, {0x49a6d74, 0x5}, ...})
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220711205747-d35199360bc9/metrics/version_parser.go:47 +0x274
k8s.io/component-base/metrics.newKubeRegistry({{0x49a6cda, 0x0}, {0x49a6cdb, 0x0}, {0x49ab0d8, 0xb}, {0x49ab0a8, 0x8}, {0x49a6d74, 0x5}, ...})
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220711205747-d35199360bc9/metrics/registry.go:320 +0x118
k8s.io/component-base/metrics.NewKubeRegistry()
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220711205747-d35199360bc9/metrics/registry.go:335 +0x78
k8s.io/component-base/metrics/legacyregistry.init()
        /root/go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/component-base@v0.0.0-20220711205747-d35199360bc9/metrics/legacyregistry/registry.go:29 +0x1d
```

--- a/Makefile
+++ b/Makefile
@@ -75,15 +75,8 @@ all: build
 require-%:
 	@if ! command -v $* 1> /dev/null 2>&1; then echo "$* not found in \$$PATH"; exit 1; fi
 
-pre-build-checks:
-ifeq ($(and $(KUBE_MAJOR_VERSION),$(KUBE_MINOR_VERSION)),)
-	$(info Kubernetes version not set. Ensure jq is installed.)
-	exit 1
-endif
-.PHONY: pre-build-checks
-
 build: WHAT ?= ./cmd/...
-build: pre-build-checks ## Build the project
+build: require-jq require-go require-git ## Build the project
 	go build $(BUILDFLAGS) -ldflags="$(LDFLAGS)" -o bin $(WHAT)
 .PHONY: build
 

